docs/src/main/asciidoc/openapi-swaggerui-guide.adoc

```diff
@@ -0,0 +1,249 @@
+import java.util.Set;
+
+@Path("/fruits")
+@Produces(MediaType.APPLICATION_JSON)
+@Consumes(MediaType.APPLICATION_JSON)
+public class FruitResource {
+
+    private Set<Fruit> fruits = Collections.newSetFromMap(Collections.synchronizedMap(new LinkedHashMap<>()));
+
+    public FruitResource() {
+        fruits.add(new Fruit("Apple", "Winter fruit"));
+        fruits.add(new Fruit("Pineapple", "Tropical fruit"));
+    }
+
+    @GET
+    public Set<Fruit> list() {
+        return fruits;
+    }
+
+    @POST
+    public Set<Fruit> add(Fruit fruit) {
+        fruits.add(fruit);
+        return fruits;
+    }
+
+    @DELETE
+    public Set<Fruit> delete(Fruit fruit) {
+        fruits.remove(fruit);
+        return fruits;
+    }
+}
+----
+
+Notice that there is no API specification or presentation specific code
+
+== Expose OpenAPI Specifications
+
+Quarkus proposes a `smallrye-openapi` extension compliant with https://github.com/eclipse/microprofile-open-api/[Eclipse MicroProfile OpenAPI]
+specification in order to generate your API's https://github.com/OAI/OpenAPI-Specification/blob/master/versions/3.0.0.md[OpenAPI v3 specification].
+
+The first and unique step is to add the `smallrye-openapi` extension to your Quarkus application:
+----
+mvn quarkus:add-extension -Dextensions="smallrye-openapi"
+----
+
+Now we are ready to run our application:
+----
+./mvnw compile quarkus:dev
+----
+
+Once your application is started, you can make a request to the default `/openapi` endpoint:
+----
+$ curl http://localhost:8080/openapi
+openapi: 3.0.1
+info:
+  title: Generated API
+  version: "1.0"
+paths:
+  /fruits:
+    get:
+      responses:
+        200:
+          description: OK
+          content:
+            application/json: {}
+    post:
+      requestBody:
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/Fruit'
+      responses:
+        200:
+          description: OK
+          content:
+            application/json: {}
+    delete:
+      requestBody:
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/Fruit'
+      responses:
+        200:
+          description: OK
+          content:
+            application/json: {}
+components:
+  schemas:
+    Fruit:
+      properties:
+        description:
+          type: string
+        name:
+          type: string
+----
+
+Hit `CTRL+C` to stop the application.
+
+== Use Swagger UI for development
+
+When building APIs, developers want to test them quickly. https://swagger.io/tools/swagger-ui/[Swagger UI] is a great tool
+permitting to visualize and interact with your APIs. It's automatically generated via your OpenAPI specification.
+
+Quarkus proposes a `swagger-ui` extension embedding a Swagger UI page configured to find your API specification at `/openapi`.
+
+*Note*: Currently the swagger-ui page is available when Quarkus is started in a dev or test mode.
```

Comment:
s/Currently the swagger-ui page is available/Currently, Swagger UI is only available/
s/in a dev or test mode/in dev or test mode/